### PR TITLE
feat: color contrast & color overlap

### PR DESCRIPTION
**Pull request by cxxxxxn**
- [x] feat: colorContrast & colorOverlap & paletteOptimization(configuration:backgroundColor)

- [x] doc: colorOverlap & colorContrast & paletteOptimization(configuration:backgroundColor)

- [x] test: color contrast & color overlap

- [x] demo: color contrast

rename: colorDistance -> colorDifference

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -8,6 +17,6 @@ export { paletteOptimization } from './optimizers';
 // get palette
 export { getPaletteFromImage, getPaletteFromString } from './extractors';
 // Professional test
```
Comment (neoddish) [inline src/index.ts:19]:
```suggestion
// evaluators
```

???

File: __tests__/unit/professionalTest/CIEDE2000.test.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Color } from '@antv/color-schema';
```
Comment (neoddish) [inline __tests__/unit/professionalTest/CIEDE2000.test.ts:1]: file path:

__tests__/unit/evaluators/....
???

> for files, you could global find 'professional' and replace...

**Review by neoddish — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (cxxxxxn) [reply src/index.ts]: The comment here is to identify the corresponding function, not the path. 

Need to modify it?

**Review by neoddish — APPROVED** [review]
LGTM

**Pull request merged**